test/gcext/Makefile

```
@@ -19,38 +19,49 @@ SRCDIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
 # get the executable suffix, if any
 EXE := $(suffix $(abspath $(JULIA)))
 
+DYLIB := .so
```

This fails on macOS. Perhaps like this?
```suggestion

OS := $(shell uname)
ifeq ($(OS), Darwin)
  DYLIB := .dylib
else
  DYLIB := .so
endif
```
Note that the top makefiles call this `SHLIB_EXT`, perhaps it'd be a good idea to imitate that?